utils/python/CIME/XML/generic_xml.py

```diff
@@ -30,47 +33,65 @@ def __init__(self, infile=None):
             # and set it's id to file
             self.filename = infile
             root = ET.Element("xml")
-            root.set('version','1.0')
-            self.root = ET.SubElement(root,"file")
-            self.root.set('id',infile)
+            root.set("version", "1.0")
+            self.root = ET.SubElement(root, "file")
+            self.root.set("id", infile)
             self.tree = ET.ElementTree(root)
 
     def read(self, infile):
         """
         Read and parse an xml file into the object
         """
-        logger.debug("read: "+infile)
+        logger.debug("read: " + infile)
         self.tree = ET.parse(infile)
         self.root = self.tree.getroot()
-        if ("version" in self.root.attrib):
+        if "version" in self.root.attrib:
             self.version = self.root.attrib["version"]
         else:
             self.version = "1.0"
         logger.debug("File version is "+self.version)
 
-
     def write(self, outfile=None):
         """
         Write an xml file from data in self
         """
-        if(outfile is None):
+        if outfile is None:
             outfile = self.filename
-        logger.debug("write: "+ outfile)
+
+        logger.debug("write: " + outfile)
         xmlstr = ET.tostring(self.root)
         doc = minidom.parseString(xmlstr)
         with open(outfile,'w') as xmlout:
             doc.writexml(xmlout,addindent='  ')
 
-
     def get_node(self, nodename, attributes=None, root=None):
```

Comment:
I was frequently confused by get_node since the name implies one, but it could actually return 0, 1, or N. This required the client to do an error-check after every call to ensure they got back the number they expected and in many cases, we forgot to do that. With this change, the API makes it explicit how what we expect and so relieves client from lots of checking and improves readability.